### PR TITLE
Adds the file-root in the hurl cli command if there is a file or Mult…

### PR DESCRIPTION
**Pull request by Xouzoura**
Simple change to allow usage of the CWD with hurl to use endpoints that require uploading files.

## WHAT

Adding the functionality to use local files of the project with hurl through the `--file-root` command.
Example

```
POST {{HOST}}/upload/{{GROUP}}/{{CONVERSATION}}/ad_hoc/upload
Authorization: Bearer {{BEARER}}
[MultipartFormData]
file: file,src/tests/files/doc1.pdf;type=application/pdf
url: ""
```
should work now. 

[Link to issue](https://github.com/jellydn/hurl.nvim/issues/195)

## WHY
Hurl does not allow by default usage of documents if the path is not specified. 
## HOW
Adding current path to the hurl if there is a file uploaded (Multipart + file must exist). However modification is needed if there is a need for documents not within the project.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] Linter
- [ ] Tests
- [ ] Review comments
- [ ] Security

----

> [!IMPORTANT]
> Adds `--file-root` option to `hurl` command for file uploads using CWD.
> 
>   - **Behavior**:
>     - Adds `--file-root` with CWD to `hurl` command if file upload is detected in `execute_hurl_cmd()` in `main.lua`.
>     - Uses `has_file_in_opts()` in `utils.lua` to check for file uploads in options.
>   - **Functions**:
>     - Adds `has_file_in_opts()` in `utils.lua` to determine if options include file uploads.
>   - **Misc**:
>     - Updates `execute_hurl_cmd()` in `main.lua` to extend command with `--file-root` when necessary.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=jellydn%2Fhurl.nvim&utm_source=github&utm_medium=referral)<sup> for 985ae22e2a264a98ef712e9b8215a40a8906e693. It will automatically update as commits are pushed.</sup>

## Summary by CodeRabbit

- **New Features**
	- Introduced a command option to specify a file root based on the current directory when a file option is included.
	- Added functionality to check for file paths in command options, enhancing error handling and validation.
	- Enhanced output handling to support JSON responses and improved logging.

- **Bug Fixes**
	- Improved error messaging for missing or inaccessible file paths in command options.

### CODE REVIEW

File: lua/hurl/utils.lua [hunk 1]
```diff
@@ -286,5 +286,30 @@ util.find_env_files_in_folders = function()
 
   return env_files
 end
+util.has_file_in_opts = function(opts)
+  if #opts == 0 then
+    print('Error: No file path provided in opts.')
```
[the remark this answers is not in the document]
Comment (jellydn) [reply lua/hurl/utils.lua:291]: Great. Thanks @Xouzoura Could you fix the review from AI review bot?

[the remark this answers is not in the document]
Comment (Xouzoura) [reply lua/hurl/utils.lua]: Done!

File: lua/hurl/main.lua [hunk 2]
```diff
@@ -190,6 +192,10 @@ local function execute_hurl_cmd(opts, callback)
 
   -- Include the HTTP headers in the output and do not colorize output.
   local cmd = vim.list_extend({ 'hurl', '-i', '--no-color' }, opts)
+  if is_file_mode then
+    local file_root = _HURL_GLOBAL_CONFIG.file_root or vim.fn.getcwd()
```
Comment (jellydn) [inline lua/hurl/main.lua:196]: Could you add the to global variable about the new config? (`file_root`) @Xouzoura

**Review by jellydn — COMMENTED** [review]
Nice works. Thanks @Xouzoura

**Review by jellydn — APPROVED** [review]
LGTM!

**Comment by jellydn** [comment]
@all-contributors add @Xouzoura for code.

**Pull request merged**